virttest/qemu_qtree.py

```diff
@@ -398,6 +398,8 @@ def parse_info_block(self, info):
         """
         additional = 0
         missing = 0
+        keys = info.keys()
+        info[re.sub("\ \(#\w+\)", "", keys[0])] = info.pop(keys[0])
```

Comment:
Hello @otubo, there could be multiple disks in the `info block` so you need to do this in a loop. Also the ` ` is restricted character in the device id, so the condition could be simpler, for example:

```diff
diff --git a/virttest/qemu_qtree.py b/virttest/qemu_qtree.py
index 8bdc6da..8b64363 100644
--- a/virttest/qemu_qtree.py
+++ b/virttest/qemu_qtree.py
@@ -398,8 +398,13 @@ class QtreeDisksContainer(object):
         """
         additional = 0
         missing = 0
-        keys = info.keys()
-        info[re.sub("\ \(#\w+\)", "", keys[0])] = info.pop(keys[0])
+        # Extract the name from "drive_image1 (#block133)"
+        for rawkey in info.keys():
+            splitkey = rawkey.split(" ", 1)
+            if len(splitkey) == 1:   # old type
+                continue
+            # new type, remove the (#block133) from the key
+            info[splitkey[0]] = info.pop(rawkey)
         for i in xrange(len(self.disks)):
             disk = self.disks[i]
             name = disk.get_qname()
```